ledger-core/virtual/execute/execute_publish_callsummary_test.go

```diff
@@ -33,21 +34,20 @@ func TestSMExecute_PublishVCallResultToCallSummarySM(t *testing.T) {
 
 		pd          = pulse.NewFirstPulsarData(10, longbits.Bits256{})
 		pulseSlot   = conveyor.NewPresentPulseSlot(nil, pd.AsRange())
-		smObjectID  = gen.UniqueLocalRefWithPulse(pd.PulseNumber)
-		smGlobalRef = reference.NewSelf(smObjectID)
+		outgoingRef = reference.NewRecordOf(gen.UniqueGlobalRef(), gen.UniqueLocalRefWithPulse(pd.PulseNumber))
```

Comment:
well would be a bit better to ensure base.PN() <= Local.PN() - here and anywhere else

Reply:
Until it is not mandatory, I'm not sure that this makes sense. Every new test need to make sure that callee and caller must be from past PN, and sometimes test does not care about PN at all. I prefere to do this kind of fix with mandatory constraint for NewRecordOf, so all the cases will be covered